src/solvers/quasi_Newton.jl

```diff
@@ -37,18 +37,20 @@ mutable struct QuasiNewtonState{
     T,
     D<:AbstractQuasiNewtonDirectionUpdate,
     SC<:StoppingCriterion,
-    S<:Stepsize,
+    S<:StepsizeStorage,
```

Comment:
I think this is breaking? Can Quasi Newton still be used with the Manopt-Armijo?

Reply:
This is only slightly breaking, someone would have to use the inner (not the outer) constructor to have their code broken by this change, or dispatch on type arguments. And there doesn't seem to be a good reason to do that. It still works for Manopt line searches because by default `StepsizeStorage` just forwards to the wrapped linesearch (only LineSearches.jl compat layer has custom behavior).

Reply:
But this does mean I can not use the Manopt line searches here? I feel that is strange.
I would prefer keeping this to be `StepSize` since I feel the storage is a decorator and hence also a stepsize? Why is that not possible? For Record/Debug in State for example it is exactly the way I wrote.

Having here a type that does not work with the line searches from this package but pony from the compat – that is a bit strange.

Reply:
Sure, I can leave the type here as `Stepsize`.